c/src/ml-api-service-training-offloading.c

```diff
@@ -734,94 +733,114 @@ _ml_service_training_offloading_set_path (ml_service_s * mls,
-    if (ret != ML_ERROR_NONE) {
-      _ml_error_report_return (ret, "Failed to start ml pipeline.");
-    }
-  } else if (training_s->type == ML_TRAINING_OFFLOADING_TYPE_RECEIVER) {
-    /* checking if all required files are received */
-    if (!_training_offloading_check_received_data (training_s)) {
-      _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
-          "Failed to receive the required data");
-    }
+  /* checking if all required files are received */
+  if (!_training_offloading_check_received_data (training_s)) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "Failed to receive the required data");
+  }
+  _training_offloading_replce_pipeline_data_path (mls);
 
-    _training_offloading_replce_pipeline_data_path (mls);
+  pipeline_node = json_from_string (training_s->receiver_pipe_json_str, NULL);
+  if (!pipeline_node) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "Failed to parse the json string, %s.",
+        training_s->receiver_pipe_json_str);
+  }
 
-    pipeline_node = json_from_string (training_s->receiver_pipe_json_str, NULL);
-    if (!pipeline_node) {
-      _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
-          "Failed to parse the json string, %s.",
-          training_s->receiver_pipe_json_str);
-    }
+  pipeline_obj = json_node_get_object (pipeline_node);
+  if (!pipeline_obj) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "Failed to get the json object from the json node.");
+  }
+  if (!json_object_has_member (pipeline_obj, "pipeline")) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "Failed to parse configuration file, cannot get the pipeline JSON object.");
+  }
 
-    pipeline_obj = json_node_get_object (pipeline_node);
-    if (!pipeline_obj) {
-      _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
-          "Failed to get the json object from the json node.");
-    }
+  pipe = json_object_get_object_member (pipeline_obj, "pipeline");
+  if (json_object_has_member (pipe, "description")) {
+    training_s->receiver_pipe =
+        g_strdup (_ml_service_get_json_string_member (pipe, "description"));
+  } else {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "Failed to parse configuration file, cannot get the pipeline description.");
+  }
 
-    if (!json_object_has_member (pipeline_obj, "pipeline")) {
-      _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
-          "Failed to parse configuration file, cannot get the pipeline JSON object.");
-    }
+  ret = ml_pipeline_construct (training_s->receiver_pipe, NULL, NULL,
+      &training_s->pipeline_h);
+  if (ML_ERROR_NONE != ret) {
+    _ml_error_report_return (ret, "Failed to construct pipeline.");
+  }
 
-    pipe = json_object_get_object_member (pipeline_obj, "pipeline");
+  ret = _training_offloading_conf_parse_pipeline (mls, pipe);
+  if (ret != ML_ERROR_NONE) {
+    return ret;
+  }
```

Comment:
I tried to modify ```ml_pipeline_construct()``` function to be used commonly, but ```ml_pipeline_construct()``` must be performed before executing ```_training_offloading_conf_parse_pipeline()```.